File: CS/HowToAccessApplicationModelInCode.Win/WinApplication.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.ExpressApp.Win;
using DevExpress.ExpressApp.Updating;
using DevExpress.ExpressApp;

namespace HowToAccessApplicationModelInCode.Win {
   public partial class HowToAccessApplicationModelInCodeWindowsFormsApplication : WinApplication {
      public HowToAccessApplicationModelInCodeWindowsFormsApplication() {
         InitializeComponent();
      }

      private void HowToAccessApplicationModelInCodeWindowsFormsApplication_DatabaseVersionMismatch(object sender, DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs e) {
         if (System.Diagnostics.Debugger.IsAttached)
         {
            e.Updater.Update();
            e.Handled = true;
         }
         else
         {
            throw new InvalidOperationException(
               "The application cannot connect to the specified database, because the latter doesn't exist or its version is older than that of the application.\r\n" +
               "The automatical update is disabled, because the application was started without debugging.\r\n" +
               "You should start the application under Visual Studio, or modify the " +
               "source code of the 'DatabaseVersionMismatch' event handler to enable automatic database update, " +
               "or manually create a database with the help of the 'DBUpdater' tool.");
         }
      }
   }
}

File: VB/HowToAccessApplicationModelInCode.Win/WinApplication.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.ExpressApp.Win
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.ExpressApp

Namespace HowToAccessApplicationModelInCode.Win
   Partial Public Class HowToAccessApplicationModelInCodeWindowsFormsApplication
	   Inherits WinApplication
	  Public Sub New()
		 InitializeComponent()
	  End Sub

	  Private Sub HowToAccessApplicationModelInCodeWindowsFormsApplication_DatabaseVersionMismatch(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs) Handles MyBase.DatabaseVersionMismatch
		  e.Updater.Update()
		  e.Handled = True
	  End Sub
   End Class
End Namespace
